src/main/kotlin/org/rust/lang/core/parser/RustParserUtil.kt

```diff
@@ -343,9 +360,17 @@ object RustParserUtil : GeneratedParserUtilBase() {
         tokenType == RsElementTypes.MACRO_EXPR &&
             '}' == b.originalText.subSequence(startOffset, endOffset).findLast { it == '}' || it == ']' || it == ')' }
 
-    private fun contextualKeyword(b: PsiBuilder, keyword: String, elementType: IElementType): Boolean {
+    private fun contextualKeyword(
+        b: PsiBuilder,
+        keyword: String,
+        elementType: IElementType,
+        nextElements: TokenSet = DEFAULT_NEXT_ELEMENTS,
+        invertNextElements: Boolean = true
```

Comment:
I think we can make the function signature much more cleaner and generic. Let's just accept a lambda that accepts next token as a parameter and return boolean. So its call site will looks like `contextualKeyword(b, "try", TRY) { it in TRY_NEXT_ELEMENTS }`